tmt/steps/provision/__init__.py

```diff
@@ -906,6 +915,10 @@ def reboot(self, hard=False, command=None, timeout=None):
                     "Seems the connection was closed too fast, ignoring.")
             else:
                 raise
+        # Wait until ssh connection drops (sleep_past_reboot is terminated)
+        self.debug(f"Waiting up to {REBOOT_TIMEOUT}s for connection to guest drop.")
+        th.join(timeout=REBOOT_TIMEOUT)
```

Comment:
What happens when `timeout` is shorter than `REBOOT_TIMEOUT`, and reboot takes much longer than `timeout`? In that case, a reboot could take longer than given `timeout`, then move to `reconnect()` with fresh `timeout`, effectively consuming way more time than expected.

I'm not sure there's something you could do about it right now, in this patch, because it seems to be a bit more complex, affecting several methods. A patch with wider scope would be needed, I'd happily work on that after #1280, I just wanted to mention there are still issues waiting to bite.

Reply:
Yeah, consuming more time than expected.  However that is something we can _improve_ later (e.g. #1389) 

Was time necessary for reboot ever considered? Some services can take a lot of time...

Reply:
How about respecting `timeout` by shortening `reconnect`(timeout - <time it took>) and  raise timeout if timeout is already over

Reply:
I'd say that would be a step in the right direction, but I for one would not take it here in this patch. There are already issues filed for honoring shorter timeouts and related troubles, I would take care of all of them later.

Reply:
Even for easy fix as e95367a33037115c07c9257ef6244ba304fc3625 ? :D

Reply:
Adjusted commit